pynapple/core/ts_group.py

```diff
@@ -1030,6 +1030,160 @@ def getby_category(self, key):
+
+        Index    rate
+        -------  ------
+              0     0.3
+             10     0.3
+
+                start    end
+        0       -5      5
+        shape: (1, 2), time unit: sec.
+
+        """
+        is_tsgroup = [isinstance(tsg, TsGroup) for tsg in tsgroups]
+        if not all(is_tsgroup):
+            not_tsgroup_index = [i+1 for i, boo in enumerate(is_tsgroup) if not boo]
+            raise TypeError(f"Passed variables at positions {not_tsgroup_index} are not TsGroup")
+
+        if len(tsgroups) == 1:
+            print('Only one TsGroup object provided, no merge needed')
+            return tsgroups[0]
+
+        tsg1    = tsgroups[0]
+        items    = tsg1.items()
+        keys     = set(tsg1.keys())
+        metadata = tsg1._metadata
+
+        for i, tsg in enumerate(tsgroups[1:]):
+            if not ignore_metadata:
+                if tsg1.metadata_columns != tsg.metadata_columns:
+                    raise ValueError(f"TsGroup at position {i+2} has different metadata columns from previous TsGroup objects. "
+                                    "Pass ignore_metadata=True to bypass")
+                metadata = pd.concat([metadata, tsg._metadata], axis=0)
+            
+            if not reset_index:
+                key_overlap = keys.intersection(tsg.keys())
+                if key_overlap:
+                    raise ValueError(f"TsGroup at position {i+2} has overlapping keys {key_overlap} with previous TsGroup objects. "
+                                     "Pass reset_index=True to bypass")
+                keys.update(tsg.keys())
+            
+            if reset_time_support:
+                time_support = None
+            else:
+                if not np.array_equal(
+                    tsg1.time_support.as_units('s').to_numpy(),
+                    tsg.time_support.as_units('s').to_numpy()
+                    ):
+                    raise ValueError(f"TsGroup at position {i+2} has different time support from previous TsGroup objects. "
+                                    "Pass reset_time_support=True to bypass")
+                time_support = tsg1.time_support
+            
+            items.extend(tsg.items())
+
+        if reset_index:
+            metadata.index = range(len(metadata))
+            data = {i: ts[1] for i, ts in enumerate(items)}
+        else:
+            data = dict(items)
+
+        if ignore_metadata:
+            return TsGroup(
+                data, time_support=time_support, bypass_check=False
+                )
+        else:
+            cols = metadata.columns.drop("rate")
+            return TsGroup(
+                data, time_support=time_support, bypass_check=False, **metadata[cols]
+                )
+    
+    def merge(self, *tsgroups, reset_index=False, reset_time_support=False, ignore_metadata=False):
+        """
+        Merge the TsGroup object with other TsGroup objects
+        See `TsGroup.merge_group` for more details
```

Comment:
since this is probably the most user facing method, I would have a nice numpydoc docstring here too!

Reply:
That's a good suggestion. Could you offer some tips on how to do so while reducing redundancy?

Reply:
I suggest to move the detailed  docstrings from the `merge_group` to the `merge`. 
In `merge_group` keep the same initial paragraph describing the method, the parameters, and the returns sections. I'll add a `See Also` section that points to `merge`.